projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -4227,9 +4227,17 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
             });
 
             diff.forEachRemovedItem((record: IterableChangeRecord<IgxColumnComponent>) => {
+                // Clear Grouping
+                this.gridAPI.clear_groupby(record.item.field);
+
                 // Clear Filtering
                 this.gridAPI.clear_filter(record.item.field);
 
+                // Close filter row
+                if (this.filteringService.filteredColumn && this.filteringService.filteredColumn.field === record.item.field) {
+                    this.filteringService.isFilterRowVisible = false;
```

Comment:
We do not need to use this.filteringRow.close(), because it checks the expression , updating chips and calls detectChanges, when we remove a column setting isFilterRowVisible to false is enough, because this column is removed and do not need to update its chips and expressions.

Reply:
The column is removed, but the filter row instance is the same for each column so it should properly clear its states. The `filteringRow.close()` method also clears some filtering service states.